### DISCUSSION TITLE
Bundling dependencies using Svelte Package

### DISCUSSION THREAD

**Discussion by akashlama1998-icloud** [sveltejs/kit#9865] [Q&A]
### Describe the problem

I have an nx monorepo where I have svelte components.. each as a package, internal and utils package separately which are just plain typescript package. The structure of my monorepo currently looks something like:-

```
.changeset/
.husky/
.vscode/
node_modules/
packages/
├ components/
│ ├ progress/
│ │ ├ dist/
│ │ ├ node_modules/
│ │ ├ src/
│ │ │ ├ circle/
│ │ │ │ ├ fill.svelte
│ │ │ │ ├ index.ts
│ │ │ │ ├ root.svelte
│ │ │ │ └ track.svelte
│ │ │ └ index.ts
│ │ ├ tests/
│ │ │ └ circle.spec.ts
│ │ ├ package.json
│ │ └ tsconfig.json
├ constants/
├ internal/
│ ├ dist/
│ ├ node_modules/
│ ├ src/
│ │ ├ create-event-forwarder.ts
│ │ └ index.ts
│ ├ tests/
│ │ └ create-event-forwarder.spec.ts
│ ├ package.json
│ └ tsconfig.json
├ utils/
│ ├ dist/
│ ├ node_modules/
│ ├ src/
│ │ ├ get-percentage.ts
│ │ └ index.ts
│ ├ tests/
│ │ └ get-percentage.spec.ts
│ ├ package.json
│ └ tsconfig.json
.eslintignore
.eslintrc.cjs
.gitignore
.npmrc
.prettierignore
.prettierrc
nx.json
package.json
pnpm-lock.yaml
pnpm-workspace.yaml
README.md
svelte.config.js
tsconfig.base.json
vitest.config.ts
```
I use svelte-package to package these individually. The package script in process, internal and utils looks like `"package": "svelte-package -i src && publint"`. So, here's my question.. process depends on function from internal and utils. Right now using the mentioned above.. I can package and publish these three packages to npm no problem. But if I wanna use the process package and install it to any other project.. right now with that I have to install both internal and utils package as well to make process package work. I was wondering is there any way I can make process package work by its own without needing to install internal and utils packages as well.

### Describe the proposed solution

I don't know what solution for that could be.. or it even is a problem or not. But it would be nice if I could implement the mentioned above. 

### Importance

would make my life easier

**Comment by elliott-with-the-longest-name-on-github** [comment]
I think this isn't a `svelte-package` thing -- you should be able to just install your workspace packages as dependencies in your other packages and have them deployed together!

Basically, if you just `pnpm i -F progress internal utils`, it should add them to your `package.json` as `dependencies` with something like `"internal": "workspace:1.0.0"`. When you publish to NPM, it should recognize these dependencies and publish them as part of your package. Are you already doing that?

**Comment by dummdidumm** [comment]
Converting to a discussion since this is a usage question. What @tcc-sejohnson proposes should work.
